grpc/stream_test.go

```diff
@@ -188,6 +189,101 @@ func TestStream_ErrorHandling(t *testing.T) {
 	)
 }
 
+// this test case is checking that everything that server sends
+// after the client finished (client.end called) is delivered to the client
+// and the end event is called
+func TestStream_ReceiveAllServerResponsesAfterEnd(t *testing.T) {
+	t.Parallel()
+
+	ts := newTestState(t)
+
+	stub := &FeatureExplorerStub{}
+
+	savedFeatures := []*grpcservice.Feature{
+		{
+			Name: "foo",
+			Location: &grpcservice.Point{
+				Latitude:  1,
+				Longitude: 2,
+			},
+		},
+		{
+			Name: "bar",
+			Location: &grpcservice.Point{
+				Latitude:  3,
+				Longitude: 4,
+			},
+		},
+	}
+
+	stub.listFeatures = func(rect *grpcservice.Rectangle, stream grpcservice.FeatureExplorer_ListFeaturesServer) error {
+		for _, feature := range savedFeatures {
+			// adding a delay to make server response "slower"
+			time.Sleep(200 * time.Millisecond)
+
+			if err := stream.Send(feature); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	grpcservice.RegisterFeatureExplorerServer(ts.httpBin.ServerGRPC, stub)
+
+	replace := func(code string) (goja.Value, error) {
```

Comment:
nit: this name isn't great, I expected it was just `ts.httpbin.Reaplace.Replace` "alias" but it also runs the code.

Additionally and (less of a nit) this should call with https://pkg.go.dev/go.k6.io/k6/js/modulestest#Runtime.RunOnEventLoop  as otherwise you are not running on the event loop - ytou can skip the `WaitOnRegisted` after.

Reply:
Yeah, after the v0.46 we have this. Let me open a separate PR for fixing this (and other tests too since they were developed before) and addressing some minor things like unexporting.

Reply:
Fix for the nit in the https://github.com/grafana/xk6-grpc/pull/50